src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -295,23 +308,23 @@ private static IntPtr GetSslMethod(bool isServer, long options)
 
             if (noSsl2 && noSsl3 && noTls11 && noTls12)
             {
-                method = libssl.TLSv1_method();
+                method = libssl.SslMethods.TLSv1_method;
             }
             else if (noSsl2 && noSsl3 && noTls10 && noTls12)
             {
-                method = libssl.TLSv1_1_method();
+                method = libssl.SslMethods.TLSv1_1_method;
             }
             else if (noSsl2 && noSsl3 && noTls10 && noTls11)
             {
-                method = libssl.TLSv1_2_method();
+                method = libssl.SslMethods.TLSv1_2_method;
             }
             else if (noSsl2 && noTls10 && noTls11 && noTls12)
             {
-                method = libssl.SSLv3_method();
+                method = libssl.SslMethods.SSLv3_method;
             }
             else
             {
-                method = libssl.SSLv23_method();
+                method = libssl.SslMethods.SSLv23_method;
```

Comment:
You could restructure this a bit to avoid checking the same conditions repeatedly, e.g.
```C#
IntPtr method = libssl.SslMethods.SSLv23_method; // default
if (noSsl2)
{
    if (noSsl3)
    {
        if (noTls11 && noTls12)
        {
            method = libssl.SslMethods.TLSv1_method;
        }
        else if (noTls10 && noTls12)
        {
            method = libssl.SslMethods.TLSv1_1_method;
        }
        else if (noTls10 && noTls11)
        {
            method = libssl.SslMethods.TLSv1_2_method;
        }
    }
    else if (noTls10 && noTls11 && noTls12)
    {
        method = libssl.SslMethods.SSLv3_method;
    }
}
```
Just something to consider.